src/main/webapp/components/directives/cache/configurationviewer.js

```diff
@@ -0,0 +1,53 @@
+(function () {
+  'use strict';
+
+  var module = angular.module('ispn.directives.cache.cacheconfiguration', ['ispn.services.utils']);
+
+  module.directive('cacheConfiguration', ['utils', function (utils) {
+      return {
+        restrict: 'E',
+        scope: {
+          currentCache: '=',
+          metadata: '=',
+          initDefaults: '=',
+          readOnly: '=',
+          cacheType: '@'
+        },
+        replace: false,
+        templateUrl: function(elem, attrs) {
+          return 'components/directives/cache/'.concat(attrs.cacheType).concat('.html');
+        },
+        link: function (scope, element, attrs) {
+
+          if (!scope.readOnly) {
+            scope.readOnly = false;
+          }
+          if (!scope.initDefaults){
+            scope.initDefaults = false;
+          }
+
+          scope.resourceDescriptionMap = {};
```

Comment:
If resourceDescriptionMap is used only inside this function (it is "private"), do not add it to scope, but make it local, like `var resourceDescriptionMap = {};`, not `scope.resourceDescriptionMap = {};`